sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/workspace.py

```diff
@@ -274,12 +379,18 @@ def _from_rest_object(cls, rest_obj: RestWorkspace) -> Optional["Workspace"]:
             managed_network=managed_network,
             feature_store_settings=feature_store_settings,
             enable_data_isolation=rest_obj.enable_data_isolation,
-            workspace_hub=rest_obj.hub_resource_id,
+            hub_id=rest_obj.hub_resource_id,
             workspace_id=rest_obj.workspace_id,
             serverless_compute=serverless_compute,
         )
 
     def _to_rest_object(self) -> RestWorkspace:
+        """Note: Unlike most entities, the create operation for workspaces does NOTE use this function,
+        and instead relies on its own internal conversion process to produce a valid ARM template.
```

Comment:
Does this function ever get used then?

If it doesn't, would it make sense to make its disuse more obvious by just having this function raise an exception?

Reply:
I don't think so, but I don't want to go through the effort of checking that today.